ecl/hql/hqlgram2.cpp

```diff
@@ -8314,16 +8314,6 @@ void HqlGram::checkJoinFlags(const attribute &err, IHqlExpression * join)
         if (join->hasAttribute(onFailAtom))
             reportError(ERR_BAD_JOINFLAG, err, "ONFAIL cannot be used in combination ONLY");
     }
-
-    IHqlExpression * rhs = join->queryChild(1);
-    if (!isKeyedJoin(join) && (rhs->getOperator() == no_filter) && (rhs != join->queryChild(0)))
```

Comment:
I would have thought the warning was still useful if it hadn't been explicitly overridden with lookup.

I was expecting a test for !join->hasAttribute(lookupAtom) (also allAtom, smartAtom, streamedAtom - probably commoning up with the list in inKeyedJoin())